src/compiler/checker.ts

```diff
@@ -4621,6 +4623,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
             if (sourceFile.symbol) {
                 if (errorNode && resolvedModule.isExternalLibraryImport && !resolutionExtensionIsTSOrJson(resolvedModule.extension)) {
+                    if (isSideEffectImport) {
+                        // This a side-effect import; ignore the implicit any.
+                        return undefined;
+                    }
                     errorOnImplicitAnyModule(/*isError*/ false, errorNode, currentSourceFile, mode, resolvedModule, moduleReference);

```

Comment:
may be let `errorOnImplicitAnyModule` take isSideEffectImport and return early instead?

Reply:
You mean return early unconditionally from this branch? Right now, it falls through on purpose. Returning early breaks a lot of tests.

Maybe I shouldn't be returning `undefined` at all, just silencing errors?

Reply:
I'm almost thinking that's true; can't users call this from the API?

Reply:
That’s what I meant .. thanks